### ISSUE TITLE
LibreSSL 2.2.1 includes OpenSSL 1.0.2's EC_curve_nid2nist API

### ISSUE THREAD

**Issue by lluixhi** [pyca/cryptography#2128]
```
src/cryptography/hazmat/bindings/__pycache__/_Cryptography_cffi_c6806e2axd5c405b7.c:1548:15: error: ‘EC_curve_nid2nist’ redeclared as different kind of symbol
 const char *(*EC_curve_nid2nist)(int) = NULL;
               ^
In file included from src/cryptography/hazmat/bindings/__pycache__/_Cryptography_cffi_c6806e2axd5c405b7.c:257:0:
/usr/include/openssl/ec.h:398:13: note: previous declaration of ‘EC_curve_nid2nist’ was here
 const char *EC_curve_nid2nist(int nid);
             ^
```
The offending bit of code is this:
```
#if defined(OPENSSL_NO_EC) || OPENSSL_VERSION_NUMBER < 0x1000200f || \
    defined(LIBRESSL_VERSION_NUMBER)
static const long Cryptography_HAS_EC_1_0_2 = 0;
const char *(*EC_curve_nid2nist)(int) = NULL;
#else
static const long Cryptography_HAS_EC_1_0_2 = 1;
#endif
```
How do we differentiate between LibreSSL >=2.2.1 and LibreSSL < 2.2.1 so this check works? They haven't been updating the version number. 2.2.1 is still unstable, so 2.1.7, at the very least, should still be supported.

**Comment by reaperhulk** [comment]
I'm not sure how we solve this short of Libre choosing to actually increment their version number...

**Comment by lluixhi** [comment]
Good news, it looks like they're going to start version bumping now:
http://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.2.2-relnotes.txt
_LibreSSL version define LIBRESSL_VERSION_NUMBER will now be bumped
   for each portable release._

**Comment by alex** [comment]
This is fixed now, right?

**Comment by reaperhulk** [comment]
Correct, fixed by #2226
